engine/src/main/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionProcessor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.resource;
+
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessor;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.StateWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedResponseWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.Writers;
+import io.camunda.zeebe.engine.state.deployment.PersistedDecision;
+import io.camunda.zeebe.engine.state.deployment.PersistedDecisionRequirements;
+import io.camunda.zeebe.engine.state.immutable.DecisionState;
+import io.camunda.zeebe.protocol.impl.record.value.deployment.DecisionRecord;
+import io.camunda.zeebe.protocol.impl.record.value.deployment.DecisionRequirementsRecord;
+import io.camunda.zeebe.protocol.impl.record.value.resource.ResourceDeletionRecord;
+import io.camunda.zeebe.protocol.record.intent.DecisionIntent;
+import io.camunda.zeebe.protocol.record.intent.DecisionRequirementsIntent;
+import io.camunda.zeebe.protocol.record.intent.ResourceDeletionIntent;
+import io.camunda.zeebe.stream.api.records.TypedRecord;
+import io.camunda.zeebe.stream.api.state.KeyGenerator;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+
+public class ResourceDeletionProcessor implements TypedRecordProcessor<ResourceDeletionRecord> {
+
+  private final StateWriter stateWriter;
+  private final TypedResponseWriter responseWriter;
+  private final KeyGenerator keyGenerator;
+  private final DecisionState decisionState;
+
+  public ResourceDeletionProcessor(
+      final Writers writers, final KeyGenerator keyGenerator, final DecisionState decisionState) {
+    stateWriter = writers.state();
+    responseWriter = writers.response();
+    this.keyGenerator = keyGenerator;
+    this.decisionState = decisionState;
+  }
+
+  @Override
+  public void processRecord(final TypedRecord<ResourceDeletionRecord> command) {
+    final var value = command.getValue();
+
+    final var drgOptional = decisionState.findDecisionRequirementsByKey(value.getResourceKey());
+    if (drgOptional.isPresent()) {
+      deleteDecisionRequirements(drgOptional.get());
+    }
```

Comment:
:thought_balloon: Just to confirm that I have the correct assumption here:

The `ResourceDeletionProcessor` will be removing BPMN, DMN, and eventually FORM resources. So the `resourceKey` may be any of these resources. As a result, we will try to locate a resource from the `decisionState`, then `processState`, and probably in the future, `formState` (or something similar), and delete the resource and associated objects that we find.

:thinking: Maybe it would be nice to implement some failure logic even now, i.e. to report to the user if no `resource` with the given `resourceKey` has been found. We can then expand this logic as support for different types of resources is added.

:question: @remcowesterhoud is there any reason you didn't add any failure messages now?

Reply:
You're assumptions are correct 👍 Because of how I did the breakdown I didn't add the rejection here yet. It's part of a separate issue https://github.com/camunda/zeebe/issues/9799